app/src/main/java/dev/beefers/vendetta/manager/ui/activity/MainActivity.kt

```diff
@@ -38,8 +40,13 @@ class MainActivity : ComponentActivity() {
             )
         }
 
-        if (preferences.installMethod == InstallMethod.SHIZUKU)
-            ShizukuPermissions.requestShizukuPermissions()
+        if (preferences.installMethod == InstallMethod.SHIZUKU) {
+            lifecycleScope.launch {
+                if (!ShizukuPermissions.waitShizukuPermissions()) {
+                    preferences.installMethod = InstallMethod.DEFAULT
```

Comment:
You should've probably shown a toast here otherwise it's changing a setting without telling the user
something like `R.string.msg_shizuku_denied`